packages/content-management/table_list_view_table/src/components/use_tag_filter_panel.tsx

```diff
@@ -45,15 +45,43 @@ export const useTagFilterPanel = ({
   addOrRemoveExcludeTagFilter,
   addOrRemoveIncludeTagFilter,
 }: Params) => {
+  const tags = useMemo(() => getTagList(), [getTagList]);
+
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
   // When the panel is "in use" it means that it is opened and the user is interacting with it.
   // When the user clicks on a tag to select it, the component is unmounted and mounted immediately, which
   // creates a new EUI transition "IN" which makes the UI "flicker". To avoid that we pass this
   // "isInUse" state which disable the transition.
   const [isInUse, setIsInUse] = useState(false);
   const [options, setOptions] = useState<TagOptionItem[]>([]);
-  const [tagSelection, setTagSelection] = useState<TagSelection>({});
-  const totalActiveFilters = Object.keys(tagSelection).length;
+  const clauseInclude: string[] = useMemo(
```

Comment:
Can we try using `id` instead of `name` for the `tagSelection`? It seems to me that it would be the cleanest approach and we wouldn't need all this logic here.